StackExchange.Profiling/Timing.cs

```diff
@@ -81,7 +81,25 @@ public Timing(MiniProfiler profiler, Timing parent, string name, decimal? minSav
         /// Gets or sets All sub-steps that occur within this Timing step. Add new children through <see cref="AddChild"/>
         /// </summary>
         [DataMember(Order = 5)]
-        public List<Timing> Children { get; set; }
+        public IList<Timing> Children
+        {
+            get
+            {
+                lock(_lockObject)
+                {
+                    return _children == null ? null : new List<Timing>(_children);
+                }
+            }
+            set { _children = value == null ? null : new List<Timing>(value); }
+        }
+
+        /// <summary>
+        /// Gets or sets All sub-steps that occur within this Timing step. Add new children through <see cref="AddChild"/>
+        /// </summary>
+        public IList<Timing> AccessChildren()
```

Comment:
I will need to add dependency on collections in order to start using iimmutablelist. I guess using that interface would make sense to emphasize that the list is not modifiable, not modifying those lists is rather pointless since i return copies.